backend/pkg/utils/helpers.go

```diff
@@ -0,0 +1,37 @@
+package utils
+
+import "go.mongodb.org/mongo-driver/bson"
+
+func CalculateNewCounter(currentScore int32, newScore int32, counter int32) int32 {
+	if currentScore == 1 {
```

Comment:
I think there is way of simplifying this: 
```
if currentScore != 0{
	return counter + (newScore - currentScore)
}

return counter + newScore
```

I just came up with it on the fly can you test if it would work?

Reply:
even better :) just: 
```
return counter + (newScore - currentScore)
```